README.MD

```diff
@@ -1,105 +1,252 @@
+# 💠 mosaic.js
 
+![Build master](https://img.shields.io/travis/OpenSTFoundation/mosaic.js/master.svg?label=build%20master&style=flat)
+![Build develop](https://img.shields.io/travis/OpenSTFoundation/mosaic.js/develop.svg?label=build%20develop&style=flat)
+![npm version](https://img.shields.io/npm/v/@openstfoundation/mosaic.js.svg?style=flat)
+[![Discuss on Discourse](https://img.shields.io/discourse/https/discuss.openst.org/topics.svg?style=flat)][discourse]
+[![Chat on Gitter](https://img.shields.io/gitter/room/OpenSTFoundation/SimpleToken.svg?style=flat)][gitter]
 
+Mosaic is a parallelization schema for decentralized applications.
+It composes heterogeneous blockchain systems into one another.
+Decentralized applications can use Mosaic to compute over a composed network of multiple blockchain systems in parallel.
 
-# Install Package
-```js
-npm install @openstfoundation/mosaic.js
-Mosaic = require('@openstfoundation/mosaic.js');
+Mosaic enables building scalable blockchain token economies through the bidirectional transposition of ERC20 tokens on one blockchain, the *origin* chain, and a utility token representation on another blockchain, the *auxiliary* chain.
+
+The protocol defines a set of actions that together perform atomic token transfers across two blockchains using gateway contracts. A gateway for a given EIP20 token is comprised of a `EIP20Gateway` contract on origin, a corresponding `EIP20CoGateway` contract on auxiliary, and and an ERC20 utility token contract on auxiliary that mints and burns utility tokens to atomically mirror tokens staked and unstaked on the origin chain.
+
+Atomicity is achieved using a 2-phase message passing architecture between the chains. Messages are declared on the source chain, and confirmed on the target chain with Patricia Merkle proofs once the source chain is finalized. Once messages are confirmed on the target chain, they can efficiently progressed with a hashlock.
+Messages can also be reverted if they are not yet completed on the target chain.
```

Comment:
But we have not provided this support in mosaic.js yet.

Reply:
```suggestion
Messages can also be reverted if they are not yet completed on the target chain.

> ⚠️ Reverting is not yet supported in `mosaic.js` and will be supported in a future version.
> In the meantime you can revert by accessing the contracts directly.
```